src/adapters/acala.ts

```diff
@@ -0,0 +1,231 @@
+import { Wallet } from "@acala-network/sdk/wallet";
+import { AnyApi, FixedPointNumber } from "@acala-network/sdk-core";
+import {
+  catchError,
+  combineLatest,
+  firstValueFrom,
+  map,
+  Observable,
+  of,
+} from "rxjs";
+
+import "@acala-network/types/argument/api-tx";
+import { ApiRx } from "@polkadot/api";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+
+import { BaseCrossChainAdapter } from "../base-chain-adapter";
+import { ChainName, chains } from "../configs";
+import { ApiNotFound } from "../errors";
+import {
+  BalanceData,
+  BasicToken,
+  CrossChainRouterConfigs,
+  CrossChainTransferParams,
+} from "../types";
+
+const ACALA_DEST_WEIGHT = "5000000000";
+
+// export const acalaRoutersConfig: Omit<CrossChainRouterConfigs, "from">[] = [
+//   {
+//     to: "interlay",
+//     token: "INTR",
+//     xcm: {
+//       fee: { token: "INTR", amount: "21787589" },
+//       weightLimit: ACALA_DEST_WEIGHT,
+//     },
+//   },
+//   {
+//     to: "interlay",
+//     token: "IBTC",
+//     xcm: {
+//       fee: { token: "IBTC", amount: "72" },
+//       weightLimit: ACALA_DEST_WEIGHT,
+//     },
+//   },
+// ];
+
+export const karuraRoutersConfig: Omit<CrossChainRouterConfigs, "from">[] = [
+  {
+    to: "kintsugi",
+    token: "KINT",
+    xcm: {
+      fee: { token: "KINT", amount: "300000000" },
+      weightLimit: ACALA_DEST_WEIGHT,
```

Comment:
This is for karura -> kintsugi? Let's set to unlimited if so

Reply:
I think this was added in the polkawallet-io bridge's acala/karura adapter for backwards compatibility.
This adapter is a copy, paste & reduced version of that.

In any event, the adapter ignores this if detects the type as `XcmV2WeightLimit` and in that case replaces this value with "Unlimited":
https://github.com/interlay/bridge/pull/51/files/fbabc145954727b14de2eb288b81e57eaf04efea#diff-8fd65d52dfb786ba16e81bd66d31c7d9023c51027f09269b29b06144a48cdf05R216

Reply:
That's quite neat, we might want do the same for our chains

Reply:
> That's quite neat, we might want do the same for our chains

Added in commit [7e58bb7](https://github.com/interlay/bridge/pull/51/commits/7e58bb7f2b7e23e6cfb0b813dd9ada0b4ed9185f)